### ISSUE TITLE
ValueError: Images of type float must be between -1 and 1.

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1691]
### Prerequisites

* [x] Did you check past issues for similar issue(s) ?
* [x] Can you reproduce the issue with any of the files found in `sct_testing_data` ?
* [x] If you are using a development environment, is your environment updated ?

### Description

Description of the issue and use the labels to add additional context.

### Steps to Reproduce

Data: **sct_testing/issue/20180424_issue1691**
~~~
sct_register_to_template -i mt1_crop.nii.gz -s mt1_crop_seg.nii.gz -l labels.nii.gz -ref subject -param step=1,type=seg,algo=slicereg,smooth=3:step=2,type=seg,algo=bsplinesyn,slicewise=1,iter=3:step=3,type=im,algo=bsplinesyn,slicewise=1,iter=3 -qc ${PATH_QC}

--
Spinal Cord Toolbox (jca_issue1690/caf11f46d92fe8765490a53325acbf0e2b36b8f1)

...

QC: layout with Axial slice
Mask type float64
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_register_to_template.py", line 686, in <module>
    main()
  File "/Users/julien/code/sct/scripts/sct_register_to_template.py", line 596, in main
    generate_qc(fname_data, fname_template2anat, fname_seg, args, os.path.abspath(param.path_qc))
  File "/Users/julien/code/sct/scripts/sct_register_to_template.py", line 618, in generate_qc
    qcslice_layout=lambda x: x.mosaic()[:2],
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 401, in add_entry
    layout(qcslice)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 209, in wrapped_f
    mask = equalized(mask)
  File "/Users/julien/code/sct/spinalcordtoolbox/reports/qc.py", line 191, in equalized
    c = skimage.exposure.equalize_adapthist(b, kernel_size=(8,8))
  File "/Users/julien/code/sct/python/lib/python2.7/site-packages/skimage/color/adapt_rgb.py", line 40, in image_filter_adapted
    return image_filter(image, *args, **kwargs)
  File "/Users/julien/code/sct/python/lib/python2.7/site-packages/skimage/exposure/_adapthist.py", line 76, in equalize_adapthist
    image = img_as_uint(image)
  File "/Users/julien/code/sct/python/lib/python2.7/site-packages/skimage/util/dtype.py", line 325, in img_as_uint
    return convert(image, np.uint16, force_copy)
  File "/Users/julien/code/sct/python/lib/python2.7/site-packages/skimage/util/dtype.py", line 205, in convert
    raise ValueError("Images of type float must be between -1 and 1.")
ValueError: Images of type float must be between -1 and 1.
~~~

**Comment by jcohenadad** [comment]
@zougloub any update on this?

**Comment by zougloub** [comment]
OK I can reproduce this with your data; for some reason the QC slice is empty. Will investigate...

**Comment by zougloub** [comment]
Interesting issue. Actually it's... a numerical error, the normalization performed was giving a maximum of 1.0000001. Your example is working when the temporary QC image is changed from float32 to float64 but I won't take a chance and just do clamping.

**Comment by zougloub** [comment]
Fixed by  #1691
